lib/ansible/modules/cloud/docker/docker_swarm_service.py

```diff
@@ -739,6 +739,23 @@ def compare(self, os):
             force_update = True
         return not differences.empty or force_update, differences, needs_rebuild, force_update
 
+    def has_publish_changed(self, old_publish):
+        for publish_item, old_publish_item in zip_longest(self.publish, old_publish):
```

Comment:
Why not return `False` directly if `len(self.publish) != len(old_publish)`, and use a regular `zip()` loop in case the lengths are equal?

Reply:
True. Didn't think about making that check before iterating. Then we can remove the None-check for each item pair too.